From c6dc32fab57dc55f266c6a7dbc0b4ccbf84f75f7 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Sat, 10 Jan 2015 03:34:51 -0800
Subject: [PATCH] Update CheckArches to fallback on Intel/ppc if ld -v doesn't
 report supported architectures

Older versions of ld64 (eg: Xcode 3.x) do not report this.

http://www.llvm.org/bugs/show_bug.cgi?id=22182

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
---
 make/platform/clang_darwin.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/make/platform/clang_darwin.mk b/make/platform/clang_darwin.mk
index 6f62170..0120547 100644
--- a/projects/compiler-rt/make/platform/clang_darwin.mk
+++ b/projects/compiler-rt/make/platform/clang_darwin.mk
@@ -20,6 +20,8 @@ CheckArches = \
         if $(LD) -v 2>&1 | grep "configured to support" \
            | tr ' ' '\n' | grep "^$$arch$$" >/dev/null 2>/dev/null; then \
           result="$$result$$arch "; \
+        elif ! ( $(LD) -v 2>&1 | grep "configured to support" ) && test "$$arch" = "i386" -o "$$arch" = "x86_64" -o "$$arch" = "ppc" -o "$$arch" = "ppc64" ; then \
+          result="$$result$$arch "; \
         else \
           printf 1>&2 \
             "warning: clang_darwin.mk: dropping arch '$$arch' from lib '$(2)'"; \
